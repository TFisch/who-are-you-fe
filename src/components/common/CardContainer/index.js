import React from 'react';
import ReincarnatedUser from '../../ReincarnatedUser';
import './styles.css';

const CardContainer = ({ users, handleClick }) => {
  const displayUsers = users.map(user => {
    return <ReincarnatedUser user={user} handleClick={handleClick} />;
  });
  return (
    <div className="card-container">
      <div className="table-headers">
        <h3 className="user-name">User Name</h3>
        <h3 className="reincarnated-person">Reincarnated As</h3>
        <p className="day">Birthday</p>
        <p className="year">Year Born</p>
        <p className="astrology-sign">Astrological Sign</p>
        <p className="delete-title">Delete?</p>
      </div>
      {displayUsers}
    </div>
  );
};

export default CardContainer;
