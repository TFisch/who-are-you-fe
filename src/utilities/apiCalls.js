export const fetchDeaths = async () => {
  const url = process.env.REACT_APP_DATABASE_API_URL + '/api/v1/deaths';
  const response = await fetch(url);
  const data = await response.json();
  await console.log(data);

};

export const fetchDateId = async day => {
  const underScoredDay = day.replace(/ /, '_');
  const url =
    process.env.REACT_APP_DATABASE_API_URL + `/api/v1/dates/${underScoredDay}`;
  const response = await fetch(url);
  const dateId = await response.json();
  return await dateId;
};

export const fetchDeathByDate = async (dateId, year) => {
  const url =
    process.env.REACT_APP_DATABASE_API_URL + `/api/v1/deaths/${dateId}/${year}`;
  const response = await fetch(url);
  const dead = await response.json();
  return await dead;
};

export const fetchUsers = async () => {
  const url = process.env.REACT_APP_DATABASE_API_URL + `/api/v1/users/`;
  const response = await fetch(url);
  const users = await response.json();
  return await users;
};

export const postUsers = async (name, death, notes) => {
  console.log(death);
  const url = process.env.REACT_APP_DATABASE_API_URL + `/api/v1/users/`;
  const response = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      name: name,
      death_id: death[0].id,
      date_id: death[0].date_id,
      notes
    })
  });
  const user = await response.json();
  return await user;
};

export const deleteUsers = async userId => {
  const url =
    process.env.REACT_APP_DATABASE_API_URL + `/api/v1/users/${userId}`;
  const response = await fetch(url, {
    method: 'DELETE'
  });
  return await response.json();
};

export const updateUser = async (userId, notes) => {
  const url =
    process.env.REACT_APP_DATABASE_API_URL + `/api/v1/users/${userId}`;
  const response = await fetch(url, {
    method: 'PATCH',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      notes
    })
  });
  return await response.json();
};
