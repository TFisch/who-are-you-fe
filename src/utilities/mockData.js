export const mockDeath = [{
  created_at: "2018-10-13T00:30:37.227Z",
  date_id: 414,
  deletable: false,
  id: 1362,
  person_name: "Lefty Gomez",
  updated_at: "2018-10-13T00:30:37.227Z",
  year: 1989
}]

export const mockUsers = [{
  astrologySign: "Aquarius",
  day: "FEBRUARY 17",
  id: 6,
  name: "Cody Taft",
  notes: "Well hello there",
  reincarnatedPerson: "Lefty Gomez",
  year: 1989
}]