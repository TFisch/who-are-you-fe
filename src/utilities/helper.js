import { fetchDateById, fetchDeathById } from './apiCalls';

export const getMonthString = month => {
  let cleanMonth;
  switch (month) {
    case '01':
      cleanMonth = 'JANUARY';
      break;
    case '02':
      cleanMonth = 'FEBRUARY';
      break;
    case '03':
      cleanMonth = 'MARCH';
      break;
    case '04':
      cleanMonth = 'APRIL';
      break;
    case '05':
      cleanMonth = 'MAY';
      break;
    case '06':
      cleanMonth = 'JUNE';
      break;
    case '07':
      cleanMonth = 'JULY';
      break;
    case '08':
      cleanMonth = 'AUGUST';
      break;
    case '09':
      cleanMonth = 'SEPTEMBER';
      break;
    case '10':
      cleanMonth = 'OCTOBER';
      break;
    case '11':
      cleanMonth = 'NOVEMBER';
      break;
    case '12':
      cleanMonth = 'DECEMBER';
      break;
    default:
      return;
  }
  return cleanMonth;
};

export const concatDates = (month, day) => {
  const cleanDate = month.concat(' ', day);
  return cleanDate;
};

export const checkDay = day => {
  let cleanDay = day;
  switch (day) {
    case '01':
      cleanDay = '1';
      break;
    case '02':
      cleanDay = '2';
      break;
    case '03':
      cleanDay = '3';
      break;
    case '04':
      cleanDay = '4';
      break;
    case '05':
      cleanDay = '5';
      break;
    case '06':
      cleanDay = '6';
      break;
    case '07':
      cleanDay = '7';
      break;
    case '08':
      cleanDay = '8';
      break;
    case '09':
      cleanDay = '9';
      break;
    default:
      return day;
  }
  return cleanDay;
};

export const userCleaner = async user => {
  const day = await fetchDateById(user.date_id);
  const deadPerson = await fetchDeathById(user.death_id);
  const cleanUser = await {
    id: user.id,
    name: user.name,
    day: day[0].day,
    reincarnatedPerson: deadPerson[0].person_name,
    year: deadPerson[0].year,
    notes: user.notes,
    astrologySign: day[0].astrology_sign
  };
  return await cleanUser;
};
