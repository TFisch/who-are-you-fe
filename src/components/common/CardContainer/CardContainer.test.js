import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './';
import { mockUsers } from '../../../utilities/mockData'

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    const users = mockUsers
    wrapper = shallow(<CardContainer users={users} />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});