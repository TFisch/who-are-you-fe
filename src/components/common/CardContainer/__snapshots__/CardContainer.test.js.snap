// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    users={
      Array [
        Object {
          "astrologySign": "Aquarius",
          "day": "FEBRUARY 17",
          "id": 6,
          "name": "Cody Taft",
          "notes": "Well hello there",
          "reincarnatedPerson": "Lefty Gomez",
          "year": 1989,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="table-headers"
        >
          <h3
            className="user-name"
          >
            User Name
          </h3>
          <h3
            className="reincarnated-person"
          >
            Reincarnated As
          </h3>
          <p
            className="day"
          >
            Birthday
          </p>
          <p
            className="year"
          >
            Year Born
          </p>
          <p
            className="astrology-sign"
          >
            Astrological Sign
          </p>
          <p
            className="delete-title"
          >
            Delete?
          </p>
        </div>,
        Array [
          <ReincarnatedUser
            user={
              Object {
                "astrologySign": "Aquarius",
                "day": "FEBRUARY 17",
                "id": 6,
                "name": "Cody Taft",
                "notes": "Well hello there",
                "reincarnatedPerson": "Lefty Gomez",
                "year": 1989,
              }
            }
          />,
        ],
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="user-name"
            >
              User Name
            </h3>,
            <h3
              className="reincarnated-person"
            >
              Reincarnated As
            </h3>,
            <p
              className="day"
            >
              Birthday
            </p>,
            <p
              className="year"
            >
              Year Born
            </p>,
            <p
              className="astrology-sign"
            >
              Astrological Sign
            </p>,
            <p
              className="delete-title"
            >
              Delete?
            </p>,
          ],
          "className": "table-headers",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "User Name",
              "className": "user-name",
            },
            "ref": null,
            "rendered": "User Name",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reincarnated As",
              "className": "reincarnated-person",
            },
            "ref": null,
            "rendered": "Reincarnated As",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Birthday",
              "className": "day",
            },
            "ref": null,
            "rendered": "Birthday",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Year Born",
              "className": "year",
            },
            "ref": null,
            "rendered": "Year Born",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Astrological Sign",
              "className": "astrology-sign",
            },
            "ref": null,
            "rendered": "Astrological Sign",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete?",
              "className": "delete-title",
            },
            "ref": null,
            "rendered": "Delete?",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClick": undefined,
          "user": Object {
            "astrologySign": "Aquarius",
            "day": "FEBRUARY 17",
            "id": 6,
            "name": "Cody Taft",
            "notes": "Well hello there",
            "reincarnatedPerson": "Lefty Gomez",
            "year": 1989,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="table-headers"
          >
            <h3
              className="user-name"
            >
              User Name
            </h3>
            <h3
              className="reincarnated-person"
            >
              Reincarnated As
            </h3>
            <p
              className="day"
            >
              Birthday
            </p>
            <p
              className="year"
            >
              Year Born
            </p>
            <p
              className="astrology-sign"
            >
              Astrological Sign
            </p>
            <p
              className="delete-title"
            >
              Delete?
            </p>
          </div>,
          Array [
            <ReincarnatedUser
              user={
                Object {
                  "astrologySign": "Aquarius",
                  "day": "FEBRUARY 17",
                  "id": 6,
                  "name": "Cody Taft",
                  "notes": "Well hello there",
                  "reincarnatedPerson": "Lefty Gomez",
                  "year": 1989,
                }
              }
            />,
          ],
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="user-name"
              >
                User Name
              </h3>,
              <h3
                className="reincarnated-person"
              >
                Reincarnated As
              </h3>,
              <p
                className="day"
              >
                Birthday
              </p>,
              <p
                className="year"
              >
                Year Born
              </p>,
              <p
                className="astrology-sign"
              >
                Astrological Sign
              </p>,
              <p
                className="delete-title"
              >
                Delete?
              </p>,
            ],
            "className": "table-headers",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "User Name",
                "className": "user-name",
              },
              "ref": null,
              "rendered": "User Name",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reincarnated As",
                "className": "reincarnated-person",
              },
              "ref": null,
              "rendered": "Reincarnated As",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Birthday",
                "className": "day",
              },
              "ref": null,
              "rendered": "Birthday",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Year Born",
                "className": "year",
              },
              "ref": null,
              "rendered": "Year Born",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Astrological Sign",
                "className": "astrology-sign",
              },
              "ref": null,
              "rendered": "Astrological Sign",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete?",
                "className": "delete-title",
              },
              "ref": null,
              "rendered": "Delete?",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": undefined,
            "user": Object {
              "astrologySign": "Aquarius",
              "day": "FEBRUARY 17",
              "id": 6,
              "name": "Cody Taft",
              "notes": "Well hello there",
              "reincarnatedPerson": "Lefty Gomez",
              "year": 1989,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    users={
      Array [
        Object {
          "astrologySign": "Aquarius",
          "day": "FEBRUARY 17",
          "id": 6,
          "name": "Cody Taft",
          "notes": "Well hello there",
          "reincarnatedPerson": "Lefty Gomez",
          "year": 1989,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="table-headers"
        >
          <h3
            className="user-name"
          >
            User Name
          </h3>
          <h3
            className="reincarnated-person"
          >
            Reincarnated As
          </h3>
          <p
            className="day"
          >
            Birthday
          </p>
          <p
            className="year"
          >
            Year Born
          </p>
          <p
            className="astrology-sign"
          >
            Astrological Sign
          </p>
          <p
            className="delete-title"
          >
            Delete?
          </p>
        </div>,
        Array [
          <ReincarnatedUser
            user={
              Object {
                "astrologySign": "Aquarius",
                "day": "FEBRUARY 17",
                "id": 6,
                "name": "Cody Taft",
                "notes": "Well hello there",
                "reincarnatedPerson": "Lefty Gomez",
                "year": 1989,
              }
            }
          />,
        ],
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="user-name"
            >
              User Name
            </h3>,
            <h3
              className="reincarnated-person"
            >
              Reincarnated As
            </h3>,
            <p
              className="day"
            >
              Birthday
            </p>,
            <p
              className="year"
            >
              Year Born
            </p>,
            <p
              className="astrology-sign"
            >
              Astrological Sign
            </p>,
            <p
              className="delete-title"
            >
              Delete?
            </p>,
          ],
          "className": "table-headers",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "User Name",
              "className": "user-name",
            },
            "ref": null,
            "rendered": "User Name",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reincarnated As",
              "className": "reincarnated-person",
            },
            "ref": null,
            "rendered": "Reincarnated As",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Birthday",
              "className": "day",
            },
            "ref": null,
            "rendered": "Birthday",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Year Born",
              "className": "year",
            },
            "ref": null,
            "rendered": "Year Born",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Astrological Sign",
              "className": "astrology-sign",
            },
            "ref": null,
            "rendered": "Astrological Sign",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete?",
              "className": "delete-title",
            },
            "ref": null,
            "rendered": "Delete?",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClick": undefined,
          "user": Object {
            "astrologySign": "Aquarius",
            "day": "FEBRUARY 17",
            "id": 6,
            "name": "Cody Taft",
            "notes": "Well hello there",
            "reincarnatedPerson": "Lefty Gomez",
            "year": 1989,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="table-headers"
          >
            <h3
              className="user-name"
            >
              User Name
            </h3>
            <h3
              className="reincarnated-person"
            >
              Reincarnated As
            </h3>
            <p
              className="day"
            >
              Birthday
            </p>
            <p
              className="year"
            >
              Year Born
            </p>
            <p
              className="astrology-sign"
            >
              Astrological Sign
            </p>
            <p
              className="delete-title"
            >
              Delete?
            </p>
          </div>,
          Array [
            <ReincarnatedUser
              user={
                Object {
                  "astrologySign": "Aquarius",
                  "day": "FEBRUARY 17",
                  "id": 6,
                  "name": "Cody Taft",
                  "notes": "Well hello there",
                  "reincarnatedPerson": "Lefty Gomez",
                  "year": 1989,
                }
              }
            />,
          ],
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="user-name"
              >
                User Name
              </h3>,
              <h3
                className="reincarnated-person"
              >
                Reincarnated As
              </h3>,
              <p
                className="day"
              >
                Birthday
              </p>,
              <p
                className="year"
              >
                Year Born
              </p>,
              <p
                className="astrology-sign"
              >
                Astrological Sign
              </p>,
              <p
                className="delete-title"
              >
                Delete?
              </p>,
            ],
            "className": "table-headers",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "User Name",
                "className": "user-name",
              },
              "ref": null,
              "rendered": "User Name",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reincarnated As",
                "className": "reincarnated-person",
              },
              "ref": null,
              "rendered": "Reincarnated As",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Birthday",
                "className": "day",
              },
              "ref": null,
              "rendered": "Birthday",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Year Born",
                "className": "year",
              },
              "ref": null,
              "rendered": "Year Born",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Astrological Sign",
                "className": "astrology-sign",
              },
              "ref": null,
              "rendered": "Astrological Sign",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete?",
                "className": "delete-title",
              },
              "ref": null,
              "rendered": "Delete?",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": undefined,
            "user": Object {
              "astrologySign": "Aquarius",
              "day": "FEBRUARY 17",
              "id": 6,
              "name": "Cody Taft",
              "notes": "Well hello there",
              "reincarnatedPerson": "Lefty Gomez",
              "year": 1989,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
